File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/enums/WebStatusCode.java
package com.zumin.sudoku.common.web.enums;

import com.zumin.sudoku.common.core.enums.StatusCode;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

/**
 * Web响应状态编码
 */
@Getter
@ToString
@AllArgsConstructor
public enum WebStatusCode implements StatusCode {

  STATISTICS_ERROR(10, "统计信息时发生错误"),
  STATISTICS_INQUIRY_DATE_INVALID(11, "查询统计信息的日期无效"),
  ;

  /**
   * 编号
   */
  private final int number;

  /**
   * 错误信息
   */
  private final String message;
}

File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/WebStatusCode.kt
package com.zumin.sudoku.common.web

import com.zumin.sudoku.common.core.code.StatusCode

/**
 * Web响应状态编码
 */
enum class WebStatusCode(private val number: Int, private val message: String) : StatusCode {
  STATISTICS_ERROR(10, "统计信息时发生错误"),
  STATISTICS_INQUIRY_DATE_INVALID(11, "查询统计信息的日期无效");

  override fun getNumber() = number

  override fun getMessage() = message
}